navpatch/navigator.go

```diff
@@ -14,25 +14,25 @@ import (
 type Navigator struct {
 	BasePath string
 	RawPatch []byte
-	BaseDir  DirTree
+	BaseDir  TreeEntry
 	Changes  map[string]*DiffStats
 }
 
-func NewNavigator(baseDir string, rawPatch []byte) (*Navigator, error) {
+func NewNavigator(r Repository, rawPatch []byte) (*Navigator, error) {
 	patchSet, err := patch.Parse(rawPatch)
 	if err != nil {
 		return nil, fmt.Errorf("parsing patch: %s", err)
 	}
 
-	tree, err := DirPathToTree(baseDir)
+	tree, err := r.GetTree()
 	if err != nil {
-		return nil, fmt.Errorf("walking directory %s: %s", baseDir, err)
+		return nil, err
```

Comment:
You should probably add a method to `Repository` that gives you the base path or URL.

Please leave the `fmt.Errorf` as it was. Adding context to the error returned from a deeper level is clarifying and done across the whole codebase.